controllers/apim/httproute_controller.go

```diff
@@ -0,0 +1,203 @@
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	gatewayapi_v1alpha2 "sigs.k8s.io/gateway-api/apis/v1alpha2"
+)
+
+const HTTPRouteNamePrefix = "hr"
+
+//+kubebuilder:rbac:groups=gateway.networking.k8s.io,resources=httproutes,verbs=get;list;watch;update;patch
+
+// HTTPRouteReconciler reconciles Gateway API's HTTPRoute object
+type HTTPRouteReconciler struct {
+	*reconcilers.BaseReconciler
+	Scheme *runtime.Scheme
+}
+
+func (r *HTTPRouteReconciler) Reconcile(eventCtx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	logger := r.Logger().WithValues("HTTPRoute", req.NamespacedName)
+	ctx := logr.NewContext(eventCtx, logger)
+
+	httproute := gatewayapi_v1alpha2.HTTPRoute{}
+	if err := r.Client().Get(ctx, req.NamespacedName, &httproute); err != nil {
+		if apierrors.IsNotFound(err) {
+			return ctrl.Result{}, nil
+		}
+		logger.Error(err, "failed to get HTTPRoute")
+		return ctrl.Result{}, err
+	}
+
+	// TODO(rahulanand16nov): handle HTTPRoute deletion for AuthPolicy
+	// check if this httproute has to be protected or not.
+	_, present := httproute.GetAnnotations()[KuadrantAuthProviderAnnotation]
+	if !present {
+		for _, parentRef := range httproute.Spec.ParentRefs {
+			gwNamespace := httproute.Namespace // consider gateway local if namespace is not given
+			if parentRef.Namespace != nil {
+				gwNamespace = string(*parentRef.Namespace)
+			}
+			gwName := string(parentRef.Name)
+			authObjKey := types.NamespacedName{
+				// Prefix is added to differentiate between routing objects
+				Name:      getAuthPolicyName(gwName, HTTPRouteNamePrefix, httproute.Name),
+				Namespace: gwNamespace,
+			}
+
+			authPolicy := istiosecurityv1beta1.AuthorizationPolicy{}
+			if err := r.Client().Get(context.Background(), authObjKey, &authPolicy); err != nil {
+				// no annotation but authpolicy exist means annotation was removed.
+				if !apierrors.IsNotFound(err) {
+					logger.Error(err, "failed to check AuthorizationPolicy existence")
+				}
+				return ctrl.Result{}, nil // this httproute is not protected.
+			}
+
+			// Orphan AuthorizationPolicy exists
+			if err := r.Client().Delete(context.Background(), &authPolicy); err != nil {
+				logger.Error(err, "failed to delete orphan authorizationpolicy")
+				return ctrl.Result{}, err
+			}
+		}
+		return ctrl.Result{}, nil
+	}
+
+	// reconcile authpolicy for the protected virtualservice
+	if err := r.reconcileAuthPolicy(ctx, logger, &httproute); err != nil {
+		logger.Error(err, "failed to reconcile AuthorizationPolicy")
+		return ctrl.Result{}, err
+	}
+	logger.Info("successfully reconciled AuthorizationPolicy")
+	return ctrl.Result{}, nil
+}
+
+func (r *HTTPRouteReconciler) reconcileAuthPolicy(ctx context.Context, logger logr.Logger, hr *gatewayapi_v1alpha2.HTTPRoute) error {
+	logger.Info("Reconciling AuthorizationPolicy")
+
+	// annotation presence is already checked.
+	providerName := hr.GetAnnotations()[KuadrantAuthProviderAnnotation]
```

Comment:
Since we are in the spike stage, I thought it best to have it configurable and can take back that flexibility later. Plus, it covers the case if the user wants some other provider other than Authorino.